src/pool/srv_target.c

```diff
@@ -973,7 +973,8 @@ ds_pool_tgt_map_update(struct ds_pool *pool, struct pool_buf *buf,
 				" map %p map_ver %u\n", DP_UUID(pool->sp_uuid),
 				pool->sp_map_version, map_version,
 				pool, pool->sp_map,
-				pool_map_get_version(pool->sp_map));
+				pool->sp_map ?
+				pool_map_get_version(pool->sp_map) : -1);
```

Comment:
Seems 'pool_map_get_version(pool->sp_map)' is  called above too, won't it cause problem? And looks the 'pool->sp_map == NULL' check is done without holding 'sp_lock', is that correct?

Reply:
this is only done on xstream 0, but it is a bit messy here, let me try to clean up a bit here.